server.js

```diff
@@ -15,40 +24,88 @@ app.use("/public", express.static(__dirname + "/public"));
 app.use(cookieParser());
 
 // get requests
-app.get("/", (req, res) => {
+app.get("/", async (req, res) => {
     const { cookies } = req
-    res.render('index', sessions[cookies.session_id]);
+    try {
+        const currentSession = await Session.findById(cookies.session_id)
+        if (currentSession === null) {
+            res.render('index');
+        } else {
+            res.render('index', currentSession);
+        }
+
+    } catch (err) {
+        res.status(500).json({ result: 'bad', error: err.message })
+    }
+
 })
 // post requests
 app.post("/signup", async (req, res) => {
+    const signupUser = await User.findOne({ username: req.body.username })
+    try {
+        if (signupUser) {
+            console.log("user not added");
+            res.redirect('/');
```

Comment:
Usually you'd want to let the user know that their username already exists, or, if you want to get *really* fancy, log them in if their password is correct.